### PR TITLE
Add confirmation dialogs for Copy Job cancel/complete actions and improve test coverage

### PR DESCRIPTION
**Pull request by BChoudhury-ms**
[Preview this branch](https://dataexplorer-preview.azurewebsites.net/pull/2323?feature.someFeatureFlagYouMightNeed=true)

This PR introduces user confirmation flows for destructive Copy Job actions and significantly strengthens test coverage and accessibility.

## Summary
Adds confirmation dialogs for Copy Job **Cancel** and **Complete** actions to prevent accidental destructive operations.

## Changes
- Introduced confirmation dialogs using `useDialog` for cancel/complete actions
- Added contextual dialog messaging and centralized strings
- Disabled all job actions while a job update is in progress
- Improved accessibility (ARIA roles, disabled states)
- Refactored and expanded unit tests for dialog flows and edge cases
- Split large container copy e2e test into focused offline/online migration tests
- Registered global `<Dialog />` component in `Main.tsx`

## Testing
- Updated unit tests for action menu and dialog flows
- Playwright tests for offline and online container copy scenarios
- Refactored the single-container Playwright tests by splitting the describe block into multiple suites, covering offline creation, online creation, and permission screen behavior.

## Screenshots
image
image
image

### CODE REVIEW

File: src/Explorer/ContainerCopy/MonitorCopyJobs/Components/CopyJobActionMenu.tsx [hunk 1]
```diff
@@ -22,6 +45,20 @@ const CopyJobActionMenu: React.FC<CopyJobActionMenuProps> = ({ job, handleClick
     return null;
   }
 
+  const showActionConfirmationDialog = (job: CopyJobType, action: CopyJobActions): void => {
+    useDialog
+      .getState()
+      .showOkCancelModalDialog(
+        ContainerCopyMessages.MonitorJobs.dialog.heading,
```
Comment (vchske) [inline src/Explorer/ContainerCopy/MonitorCopyJobs/Components/CopyJobActionMenu.tsx:52]: Will this set the heading to empty ""?

File: test/sql/containercopy/offlineMigration.spec.ts [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+    await expect(createCopyJobButton).toBeVisible();
+    await createCopyJobButton.click();
+    panel = frame.getByTestId("Panel:Create copy job");
+    await expect(panel).toBeVisible();
+
+    // Reduced wait time for better performance
+    await page.waitForTimeout(2000);
+
+    // Setup subscription and account
+    const subscriptionDropdown = panel.getByTestId("subscription-dropdown");
```
[the remark this answers is not in the document]
Comment (BChoudhury-ms) [reply test/sql/containercopy/offlineMigration.spec.ts:53]: This is intentional.

File: test/sql/containercopy/onlineMigration.spec.ts [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+    await expect(createCopyJobButton).toBeVisible();
+    await createCopyJobButton.click();
+    panel = frame.getByTestId("Panel:Create copy job");
+    await expect(panel).toBeVisible();
+
+    // Reduced wait time for better performance
+    await page.waitForTimeout(1000);
+
+    // Enable online migration mode
+    const migrationTypeContainer = panel.getByTestId("migration-type");
```
[the remark this answers is not in the document]
Comment (BChoudhury-ms) [reply test/sql/containercopy/onlineMigration.spec.ts:47]: This is intentional

File: test/sql/containercopy/onlineMigration.spec.ts [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+
+    const copyButton = panel.getByRole("button", { name: "Copy", exact: true });
+
+    const copyJobCreationPromise = waitForApiResponse(
+      page,
+      `${targetAccountName}/dataTransferJobs/${onlineMigrationJobName}`,
+      "PUT",
+    );
+    await copyButton.click();
+    await page.waitForTimeout(1000); // Reduced wait time
```
[the remark this answers is not in the document]
Comment (BChoudhury-ms) [reply test/sql/containercopy/onlineMigration.spec.ts:117]: This is intentional

File: test/sql/containercopy/offlineMigration.spec.ts [hunk 5]
```diff
@@ -0,0 +1,258 @@
...
+  let targetAccountName: string;
+  let expectedSubscriptionName: string;
+  let expectedCopyJobNameInitial: string;
+
+  test.beforeEach("Setup for offline migration test", async ({ browser }) => {
+    await createMultipleTestContainers({ accountType: TestAccount.SQLContainerCopyOnly, containerCount: 2 });
+
+    page = await browser.newPage();
+    ({ wrapper, frame } = await ContainerCopy.open(page, TestAccount.SQLContainerCopyOnly));
+    expectedJobName = `offline_test_job_${Date.now()}`;
```
[the remark this answers is not in the document]
Comment (BChoudhury-ms) [reply test/sql/containercopy/offlineMigration.spec.ts:29]: Will update in future if anytime it failed in the PR pipeline

File: src/Explorer/ContainerCopy/MonitorCopyJobs/Components/CopyJobActionMenu.test.tsx [hunk 6]
```diff
@@ -1,10 +1,25 @@
+/* eslint-disable jest/no-conditional-expect */
```
[the remark this answers is not in the document]
Comment (BChoudhury-ms) [reply src/Explorer/ContainerCopy/MonitorCopyJobs/Components/CopyJobActionMenu.test.tsx:1]: Not required

File: src/Explorer/ContainerCopy/ContainerCopyMessages.ts [hunk 7]
```diff
@@ -184,5 +184,10 @@ export default {
       Skipped: "Cancelled",
       Cancelled: "Cancelled",
     },
+    dialog: {
+      heading: "",
```
[the remark this answers is not in the document]
Comment (BChoudhury-ms) [reply src/Explorer/ContainerCopy/ContainerCopyMessages.ts:188]: This is intentional

File: src/Explorer/ContainerCopy/MonitorCopyJobs/Components/CopyJobActionMenu.tsx (hunk 1 above)
Comment (BChoudhury-ms) [reply src/Explorer/ContainerCopy/MonitorCopyJobs/Components/CopyJobActionMenu.tsx:52]: Yes, this does set to empty "". _[Sample screenshot with heading]_
image

File: src/Explorer/ContainerCopy/MonitorCopyJobs/Components/CopyJobActionMenu.tsx [hunk 8]
```diff
@@ -67,7 +104,7 @@ const CopyJobActionMenu: React.FC<CopyJobActionMenuProps> = ({ job, handleClick
           key: CopyJobActions.complete,
           text: ContainerCopyMessages.MonitorJobs.Actions.complete,
           iconProps: { iconName: "CheckMark" },
-          onClick: () => handleClick(job, CopyJobActions.complete, setUpdatingJobAction),
+          onClick: () => showActionConfirmationDialog(job, CopyJobActions.complete),
           disabled: isThisJobUpdating && updatingAction === CopyJobActions.complete,
```
[the remark this answers is not in the document]
Comment (BChoudhury-ms) [reply src/Explorer/ContainerCopy/MonitorCopyJobs/Components/CopyJobActionMenu.tsx:108]: will accommodate in upcoming PR

**Review by nishthaAhujaa — APPROVED** [review]
